src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,39 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
 import { Loader } from '../Loader';
+import { TodoModalProps } from './TodoModal.types';
+import { getUser } from '../../utils/getUser';
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [userName, setUserName] = useState('');
+  const [userIsLoading, setUserIsLoading] = useState(false);
+  const targetUserId = selectedTodo?.userId;
+
+  useEffect(() => {
+    if (targetUserId) {
+      setUserIsLoading(true);
+      getUser(targetUserId)
+        .then(user => {
+          setUserName(user.name);
+        })
+        .finally(() => {
+          setUserIsLoading(false);
+        });
+    }
+  }, []);
```

Comment:
just add it to the dependency array:
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/86923089/7e186928-5434-4072-b60d-b74a2ad74125)